transmission_interface/test/simple_transmission_loader_test.cpp

```diff
@@ -0,0 +1,382 @@
+        <param name="example_param_read_for_sec">3</param>
+      </hardware>
+      <joint name="joint2">
+        <command_interface name="velocity">
+          <param name="min">-1</param>
+          <param name="max">1</param>
+        </command_interface>
+        <state_interface name="velocity"/>
+      </joint>
+    </ros2_control>
+    <ros2_control name="RRBotModularPositionSensorJoint1" type="sensor">
+      <hardware>
+        <plugin>ros2_control_demo_hardware/PositionSensorHardware</plugin>
+        <param name="example_param_read_for_sec">2</param>
+      </hardware>
+      <joint name="joint1">
+        <state_interface name="position"/>
+      </joint>
+    </ros2_control>
+    <ros2_control name="RRBotModularPositionSensorJoint2" type="sensor">
+      <hardware>
+        <plugin>ros2_control_demo_hardware/PositionSensorHardware</plugin>
+        <param name="example_param_read_for_sec">2</param>
+      </hardware>
+      <joint name="joint2">
+        <state_interface name="position"/>
+      </joint>
+    </ros2_control>
+  </robot>
+  )";
+
+  std::vector<hardware_interface::HardwareInfo> infos =
+    hardware_interface::parse_control_resources_from_urdf(urdf_to_test);
+  ASSERT_EQ(1lu, infos[0].transmissions.size());
+
+  // Transmission loader
+  TransmissionPluginLoader loader;
+  std::shared_ptr<transmission_interface::TransmissionLoader> transmission_loader =
+    loader.create(infos[0].transmissions[0].type);
+  ASSERT_TRUE(nullptr != transmission_loader);
+
+  std::shared_ptr<transmission_interface::Transmission> transmission;
+  const hardware_interface::TransmissionInfo & info = infos[0].transmissions[0];
+  transmission = transmission_loader->load(info);
+  ASSERT_TRUE(nullptr != transmission);
+  ASSERT_STREQ(infos[0].transmissions[0].joints[0].role.c_str(), "joint1");
+
+  // Validate transmission
+  transmission_interface::SimpleTransmission * simple_transmission =
+    dynamic_cast<transmission_interface::SimpleTransmission *>(transmission.get());
+  ASSERT_TRUE(nullptr != simple_transmission);
+  EXPECT_EQ(325.949, simple_transmission->get_actuator_reduction());
+  EXPECT_EQ(0.0, simple_transmission->get_joint_offset());
+}
+
+TEST(SimpleTransmissionLoaderTest, MinimalSpec)
+{
+  std::string urdf_to_test = R"(
+ <?xml version="1.0"?>
+
+  <robot name="robot" xmlns="http://www.ros.org">
+  <ros2_control name="MinimalSpec" type="actuator">
+    <joint name="joint1">
+        <command_interface name="velocity">
+          <param name="min">-1</param>
+          <param name="max">1</param>
+        </command_interface>
+        <state_interface name="velocity"/>
+      </joint>
+      <transmission name="transmission1">
+        <plugin>transmission_interface/SimpleTransmission</plugin>
+        <joint name="joint1" role="joint1">
+          <mechanical_reduction>50</mechanical_reduction>
+        </joint>
+      </transmission>
+  </ros2_control>
+  </robot>
```

Comment:
This has to be moved to the `assets` repo, but anyway let's be more consistent with indentation:

```suggestion
<?xml version="1.0"?>

<robot name="robot" xmlns="http://www.ros.org">
  <ros2_control name="MinimalSpec" type="actuator">
    <joint name="joint1">
      <command_interface name="velocity">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <state_interface name="velocity"/>
    </joint>
    <transmission name="transmission1">
      <plugin>transmission_interface/SimpleTransmission</plugin>
      <joint name="joint1" role="joint1">
        <mechanical_reduction>50</mechanical_reduction>
      </joint>
    </transmission>
  </ros2_control>
</robot>
```